benchclients/python/benchclients/base.py

```diff
@@ -0,0 +1,75 @@
+import abc
+from json import dumps
+from typing import Optional
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from .logging import log
+
+
+class BaseClient(abc.ABC):
+    """A client to interact with an API.
+    Parameters
```

Comment:
nit: you need a newline above this one, else the RST won't render correctly if we ever render it